src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TryStatement.cs

```diff
@@ -43,7 +45,7 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
         /// such as throwing an exception? This implementation is conservative, in the sense
         /// that it may return true when the statement actually may have no side effects.
         /// </summary>
-        private static bool HasSideEffects(BoundStatement statement)
+        private static bool HasSideEffects(BoundStatement? statement)
```

Comment:
Consider annotated with `NotNullWhen(true)`

Reply:
I could not find any client who would benefit from that.

---
In reply to: [390545649](https://github.com/dotnet/roslyn/pull/42068#discussion_r390545649) [](ancestors = 390545649)

Reply:
> I could not find any client who would benefit from that.

The point is not about today's clients. Someone might benefit from this in the future, and the condition is already met by the implementation.